src/controls/movement-controls.js

```diff
@@ -118,21 +118,34 @@ AFRAME.registerComponent('movement-controls', {
       }
 
       if (data.constrainToNavMesh
-          && velocityCtrl.isNavMeshConstrained !== false) {
+          && (!velocityCtrl || velocityCtrl.isNavMeshConstrained !== false)) {
 
         if (velocity.lengthSq() < EPS) return;
 
-        start.copy(el.object3D.position);
-        end
-          .copy(velocity)
-          .multiplyScalar(dt / 1000)
-          .add(start);
+        const cameraEl = data.camera;
+        const cameraOffset = new THREE.Vector3();
+        cameraEl.object3D.getWorldPosition(cameraOffset);
+        if (el.object3D.parent) {
+          el.object3D.parent.worldToLocal(cameraOffset);
+        }
+        cameraOffset.sub(el.object3D.position);
+        cameraOffset.y = 0;
```

Comment:
Do that extra calculation only if this.el.sceneEl.is("vr-mode")
When not in VR, the code should do what we had before.